### ISSUE TITLE
Ruff does not report the `assignment-from-no-return` error

### ISSUE THREAD

**Issue by behnazh-w** [astral-sh/ruff#9397]
I was expecting ruff to report an error similar to Pylint's [assignment-from-no-return (E1111)](https://pylint.pycqa.org/en/v3.0.3/user_guide/checkers/features.html#typecheck-checker-messages) for the following code, however, I get no errors.
```python
def print_sorted_list() -> None:
    """Print a sorted list."""
    sorted_list = [1, 2, 3].sort() # `list.sort()` is an in-place function and doesn't return anything.
    print(sorted_list)
```

The command I invoked:
```bash
ruff /path/to/file.py
```
The current Ruff settings (any relevant sections from your `pyproject.toml`).
```toml
[tool.ruff]
select = [
    "C4",
    "E",
    "F",
    "W",
    "B",
    "I",
    "G",
    "RUF",
    "UP",
    "ISC",
    "PERF",
    "PL",
]
```
The current Ruff version (`ruff --version`):
```
ruff 0.1.11
```

The error I get by running Pylint v3.0.3:
```
E1111: Assigning result of a function call, where the function has no return (assignment-from-no-return)
```

**Comment by tjkuson** [comment]
I don't think that Pylint rule hasn't been implemented yet. You can view the progress of which Pylint rules have been implemented here: #970.

**Comment by behnazh-w** [comment]
Thanks for the reference. Closing this issue as a duplicate of https://github.com/astral-sh/ruff/issues/970.
